Source/Core/DolphinWX/Frame.cpp

```diff
@@ -1416,29 +1416,73 @@ void CFrame::ParseHotkeys()
 	{
 		State::Load(g_saveSlot);
 	}
-	if (IsHotkey(HK_DECREASE_DEPTH, true))
+
+	auto savePreset = [](const std::string& param, int value)
+	{
+		IniFile localIni = SConfig::GetInstance().LoadLocalGameIni();
+		localIni.GetOrCreateSection("Enhancements")->Set(
+			StringFromFormat("Stereo%s_%d", param.c_str(), g_Config.iStereoActivePreset),
+			value);
+		std::string iniFileName = File::GetUserPath(D_GAMESETTINGS_IDX) + SConfig::GetInstance().GetUniqueID() + ".ini";
+		OSD::AddMessage(StringFromFormat("%s: %d", param.c_str(), value) , 1000);
+		localIni.Save(iniFileName);
+	};
+
+	if (IsHotkey(HK_DECREASE_DEPTH))
 	{
 		if (--g_Config.iStereoDepth < 0)
 			g_Config.iStereoDepth = 0;
+		g_Config.oStereoPresets[g_Config.iStereoActivePreset].depth = g_Config.iStereoDepth;
+		savePreset("Depth", g_Config.iStereoDepth);
 	}
-	if (IsHotkey(HK_INCREASE_DEPTH, true))
+	if (IsHotkey(HK_INCREASE_DEPTH))
 	{
 		if (++g_Config.iStereoDepth > 100)
 			g_Config.iStereoDepth = 100;
+		g_Config.oStereoPresets[g_Config.iStereoActivePreset].depth = g_Config.iStereoDepth;
+		savePreset("Depth", g_Config.iStereoDepth);
 	}
-	if (IsHotkey(HK_DECREASE_CONVERGENCE, true))
+	if (IsHotkey(HK_DECREASE_CONVERGENCE))
 	{
 		g_Config.iStereoConvergence -= 5;
 		if (g_Config.iStereoConvergence < 0)
 			g_Config.iStereoConvergence = 0;
+		g_Config.oStereoPresets[g_Config.iStereoActivePreset].convergence = g_Config.iStereoConvergence;
+		savePreset("Convergence", g_Config.iStereoConvergence);
 	}
-	if (IsHotkey(HK_INCREASE_CONVERGENCE, true))
+	if (IsHotkey(HK_INCREASE_CONVERGENCE))
 	{
 		g_Config.iStereoConvergence += 5;
 		if (g_Config.iStereoConvergence > 500)
 			g_Config.iStereoConvergence = 500;
+		g_Config.oStereoPresets[g_Config.iStereoActivePreset].convergence = g_Config.iStereoConvergence;
+		savePreset("Convergence", g_Config.iStereoConvergence);
 	}
 
+	if (IsHotkey(HK_SWITCH_STEREOSCOPY_PRESET))
```

Comment:
What shall this hotkey do? Switch to 0 if the current preset is != 0? Why shall it switch to exactly 1 if the current present is 0? I think this hotkey is a bit redundant.